src/main/java/pl/edu/mimuw/chess/Goniec.java

```diff
@@ -0,0 +1,105 @@
+package pl.edu.mimuw.chess;
+
+import pl.edu.mimuw.Figura;
+
+import java.util.ArrayList;
+
+public class Goniec implements Figura {
+  public Kolor kolor;
+  public V2 pozycja;
+
+
+  public Goniec(Kolor kolor, V2 pozycja) {
+    this.pozycja = pozycja;
+    this.kolor = kolor;
+  }
+
+
+  @Override
+  public String reprezentacja() {
+    if (this.kolor == Kolor.BIALY)
+    return "♗";
+    return "♝";
+  }
+
+  @Override
+  public String toString() {
+    return this.reprezentacja();
+  }
+
+  @Override
+  public V2 getPozycja() {
+    return this.pozycja;
+  }
+
+  @Override
+  public Kolor kolor() {
+    return this.kolor;
+  }
+
+  @Override
+  public ArrayList<Ruch> wszystkieRuchy() {
+    ArrayList<Ruch> res = new ArrayList<>();
+    V2 poz = new V2(this.pozycja.x-1, this.pozycja.y-1);
```

Comment:
`V2 poz = this.pozycja.add(new V2(-1, -1));` a najlepiej to miec w kodzie gdzies np w klasie `V2`
`public static final V2 SW = new V2(-1, -2); // SW od kierunku świata`
`V2 poz = this.pozycja.add(SW);`